tests/containers/container_diff.pm

```diff
@@ -0,0 +1,52 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Print and save diffs between two cotaniners using container-diff tool
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
+
+use base 'consoletest';
+use testapi;
+use utils;
+use strict;
+use warnings;
+use containers::container_images;
+use suse_container_urls 'get_suse_container_urls';
+
+sub perform_container_diff {
+    my $first  = $_[0];
+    my $second = $_[1];
+    die "You must define two images!" unless $second;
+
+    # The container-diff is available only for x86_64 architecture
+    return unless check_var('ARCH', 'x86_64');
+
+    zypper_call("install container-diff") if (script_run("which container-diff") != 0);
+
+    # container-diff
+    my $image_file = $first =~ s/\/|:/-/gr;
+    if (script_run("docker pull $second", 600) == 0) {
```

Comment:
This only downloads `$second` which means that `$first` has to be present already. As this is now its own module, it's no longer obvious why.

Reply:
True! The `container-diff` does not require `docker` any more and is able to download the image itself. Thanks